x/config/config.go

```diff
@@ -45,25 +133,17 @@ func NewStreamDialer(transportConfig string) (dialer transport.StreamDialer, err
 }
 
 func newStreamDialerFromPart(innerDialer transport.StreamDialer, oneDialerConfig string) (transport.StreamDialer, error) {
-	oneDialerConfig = strings.TrimSpace(oneDialerConfig)
-
-	if oneDialerConfig == "" {
-		return nil, errors.New("empty config part")
-	}
-
-	url, err := url.Parse(oneDialerConfig)
+	url, err := parseConfigPart(oneDialerConfig)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse config part: %w", err)
```

Comment:
We already included the error message "failed to parse config part" in `parseConfigPart`.
```suggestion
		return nil, err
```

Reply:
Updated `parseConfigPart` to be more specific, so we can keep this one.